### PR TITLE
[SPARK-42733][CONNECT][PYTHON] Fix DataFrameWriter.save to work without path parameter

### PR DESCRIPTION
**Pull request by ueshin**
### What changes were proposed in this pull request?

Fixes `DataFrameWriter.save` to work without path parameter.

### Why are the changes needed?

`DataFrameWriter.save` should work without path parameter because some data sources, such as jdbc, noop, works without those parameters.

```py
>>> print(spark.range(10).write.format("noop").mode("append").save())
Traceback (most recent call last):
...
AssertionError: Invalid configuration of WriteCommand, neither path or table present.
```

### Does this PR introduce _any_ user-facing change?

The data sources that don't need path parameter will work.

```py
>>> print(spark.range(10).write.format("noop").mode("append").save())
None
```

### How was this patch tested?

Added a test.

### CODE REVIEW

**Review by hvanhovell — APPROVED** [review]
LGTM

File: connector/connect/server/src/test/scala/org/apache/spark/sql/connect/planner/SparkConnectProtoSuite.scala [hunk 1]
```diff
@@ -554,7 +554,8 @@ class SparkConnectProtoSuite extends PlanTest with SparkConnectPlanTest {
       parameters = Map("columnName" -> "`duplicatedcol`"))
   }
 
-  test("Writes fails without path or table") {
+  // TODO(SPARK-42733): Writes without path or table should work.
+  ignore("Writes fails without path or table") {
```
Comment (ueshin) [inline connector/connect/server/src/test/scala/org/apache/spark/sql/connect/planner/SparkConnectProtoSuite.scala:558]: @zhenlineo May I ask you to fix this test later?

Comment (zhenlineo) [reply connector/connect/server/src/test/scala/org/apache/spark/sql/connect/planner/SparkConnectProtoSuite.scala:558]: Yeah, sure. Thanks

**Review by amaliujia — APPROVED** [review]
LGTM

**Review by HyukjinKwon — APPROVED** [review]

**Comment by zhengruifeng** [comment]
merged to master/branch-3.4

File: python/pyspark/sql/tests/test_datasources.py [hunk 2]
```diff
@@ -192,6 +193,23 @@ def test_ignore_column_of_all_nulls(self):
         finally:
             shutil.rmtree(path)
 
+    def test_jdbc(self):
+        db = f"memory:{uuid.uuid4()}"
```
Comment (zhenlineo) [inline python/pyspark/sql/tests/test_datasources.py:197]: @ueshin What kind of config do I need to make this work? I got error: Database xxx not found.

Comment (ueshin) [reply python/pyspark/sql/tests/test_datasources.py:197]: I think it should work without any configs.
Could you try to rebuild?

Comment (zhenlineo) [reply python/pyspark/sql/tests/test_datasources.py:197]: @ueshin Yeah, the python test works. I want to do the same for scala client, but I am missing some spark config?

Comment (ueshin) [reply python/pyspark/sql/tests/test_datasources.py:197]: Does #40358 include the test? I can try it and see.

Comment (zhenlineo) [reply python/pyspark/sql/tests/test_datasources.py:197]: Oh I missed "create", thanks. Works for scala client too!

**Pull request closed without merging**